packages/firestore/scripts/api-report.ts

```diff
@@ -0,0 +1,167 @@
+    encoding: 'utf-8'
+  });
+}
+
+function loadApiExtractorConfig(
+  typescriptDtsPath: string,
+  rollupDtsPath: string,
+  dtsRollupEnabled: boolean,
+  apiReportEnabled: boolean
+): ExtractorConfig {
+  const apiExtractorJsonPath = path.resolve(tmpDir, 'api-extractor.json');
+  const apiExtractorJson = {
+    extends: baseApiExtractorConfigFile,
+    mainEntryPointFilePath: typescriptDtsPath,
+    'dtsRollup': {
+      'enabled': dtsRollupEnabled,
+      publicTrimmedFilePath: rollupDtsPath
+    },
+    'tsdocMetadata': {
+      'enabled': false
+    },
+    'apiReport': {
+      'enabled': apiReportEnabled,
+      reportFolder
+    },
+    'messages': {
+      'extractorMessageReporting': {
+        'ae-missing-release-tag': {
+          'logLevel': 'none'
+        },
+        'ae-unresolved-link': {
+          'logLevel': 'none'
+        },
+        'ae-forgotten-export': {
+          'logLevel': apiReportEnabled ? 'error' : 'none'
+        }
+      },
+      'tsdocMessageReporting': {
+        'tsdoc-undefined-tag': {
+          'logLevel': 'none'
+        }
+      }
+    }
+  };
+  fs.writeFileSync(apiExtractorJsonPath, JSON.stringify(apiExtractorJson), {
+    encoding: 'utf-8'
+  });
+  return ExtractorConfig.loadFileAndPrepare(apiExtractorJsonPath);
+}
+
+export async function generateApi(sdkVariant: 'exp' | 'lite'): Promise<void> {
+  // The .d.ts file generated by the Typescript compiler as we transpile our
+  // sources
+  const typescriptDtsPath: string = path.resolve(
+    __dirname,
+    `../dist/firestore/${sdkVariant}/index.d.ts`
+  );
+  // A "bundled" version of our d.ts files that includes all public and private
+  // types
+  const rollupDtsPath: string = path.resolve(
+    __dirname,
+    `../dist/${sdkVariant}/private.d.ts`
+  );
+  // A customer-facing d.ts file that only include the public APIs
+  const publicDtsPath: string = path.resolve(
+    __dirname,
+    `../dist/${sdkVariant}/index.d.ts`
+  );
+
+  console.log(
+    `Running API Extractor configuration for firestore-${sdkVariant}...`
+  );
+  writeTypescriptConfig();
+  writePackageJson(sdkVariant);
+
+  let extractorConfig = loadApiExtractorConfig(
+    typescriptDtsPath,
+    rollupDtsPath,
+    /* dtsRollupEnabled= */ true,
+    /* apiReportEnabled= */ false
+  );
+  Extractor.invoke(extractorConfig, {
+    localBuild: true
+  });
+
+  console.log('Generated rollup DTS');
+  pruneDts(rollupDtsPath, publicDtsPath);
+  console.log('Pruned DTS file');
+  await removeUnusedImports(publicDtsPath);
+  console.log('Removed unused imports');
+
+  extractorConfig = loadApiExtractorConfig(
+    publicDtsPath,
+    rollupDtsPath,
+    /* dtsRollupEnabled= */ false,
+    /* apiReportEnabled= */ true
+  );
+  Extractor.invoke(extractorConfig, { localBuild: true });
+  console.log(
+    `API report for firestore-${sdkVariant} written to ${reportFolder}`
+  );
+}
+
+void generateApi('lite').then(() => generateApi('exp'));
```

Comment:
What does void do?

Reply:
It suppresses `no-floating-promises`: https://github.com/palantir/tslint/issues/3020